frontend/src/test/java/com/google/tripmeout/serialization/GsonTripModelTypeAdapterTest.java

```diff
@@ -0,0 +1,87 @@
+package com.google.tripmeout.serialization;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParseException;
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.serialization.GsonTripModelTypeAdapter;
+import com.google.tripmeout.serialization.testdata.TestDataAccessUtil;
+import java.util.UUID;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GsonTripModelTypeAdapterTest {
```

Comment:
Please add serialization tests as well (via round trip tests). This is especially important since (at least as of now) we're _only_ using JSON serialization on the server.